### PR TITLE
add timeout for exec interface

### PR DESCRIPTION
**Pull request by zhouhaibing089**
This should get us away from situations like https://github.com/kubernetes/kubernetes/issues/63331.

A little bit more context, the `os/exec` package starts to accept `context.Context` in golang 1.7. We should leverage that so we can have a more predictable behavior, then. 

```release-note
NONE
```

### CODE REVIEW

File: pkg/util/iptables/iptables.go [hunk 1]
```diff
@@ -413,11 +415,18 @@ func iptablesCommand(protocol Protocol) string {
...
+func (runner *runner) runContext(ctx context.Context, op operation, args []string) ([]byte, error) {
 	iptablesCmd := iptablesCommand(runner.protocol)
 	fullArgs := append(runner.waitFlag, string(op))
 	fullArgs = append(fullArgs, args...)
 	glog.V(5).Infof("running iptables %s %v", string(op), args)
-	return runner.exec.Command(iptablesCmd, fullArgs...).CombinedOutput()
+	if ctx == nil {
+		return runner.exec.Command(iptablesCmd, fullArgs...).CombinedOutput()
+	}
+	return runner.exec.CommandContext(ctx, iptablesCmd, fullArgs...).CombinedOutput()
```
Comment (dims) [inline pkg/util/iptables/iptables.go]: Should we not check if we timed out? `ctx.Err() == context.DeadlineExceeded`

Comment (zhouhaibing089) [reply pkg/util/iptables/iptables.go]: @dims: We should, but in the place where the context object is constructed, see L504 in the same file.

Comment (dims) [reply pkg/util/iptables/iptables.go]: Ah thanks! @zhouhaibing089

**Comment by dims** [comment]
@zhouhaibing089 do we need to fix the code here first? https://github.com/kubernetes/utils/tree/master/exec (and then vendor it back in?)

**Comment by zhouhaibing089** [comment]
@dims: Oh, I thought `k8s.io/utils` is another staging repo. Will do!

**Comment by zhouhaibing089** [comment]
Well, so https://github.com/kubernetes/utils/pull/27 already have this.

**Comment by dims** [comment]
@zhouhaibing089 ah cool, let's update vendorized to latest code from that repo (and update Godeps.json etc...)

File: Godeps/Godeps.json [hunk 2]
```diff
@@ -2363,82 +2363,82 @@
 		},
 		{
 			"ImportPath": "github.com/opencontainers/runc/libcontainer",
-			"Comment": "v1.0.0-rc5-46-g871ba2e5",
+			"Comment": "v1.0.0-rc5-46-g871ba2e",
```
Comment (dims) [inline Godeps/Godeps.json]: we did not update this component, can you please change the SHA back to original? Looks like your environment chopped off the last digit

**Comment by dims** [comment]
@zhouhaibing089 lot of BUILD files were removed ... can you please add them back?

File: Godeps/Godeps.json (hunk 2 above)
Comment (zhouhaibing089) [reply Godeps/Godeps.json]: @dims: I am not sure which part I did wrong, I followed the steps at https://github.com/kubernetes/community/blob/master/contributors/devel/godep.md which is pretty helpful.

1.  ./hack/run-in-gopath.sh godep restore
1.  ./hack/run-in-gopath.sh go get k8s.io/utils/exec
1.  ./hack/run-in-gopath.sh godep save

The stripping of last digt might be related with which part of my environment, git version?

Comment (cblecker) [reply Godeps/Godeps.json]: Yes, please update your git version.

**Pull request merged**